src/groups/bmq/bmqp/bmqp_crc32c.cpp

```diff
@@ -573,20 +576,25 @@ static inline unsigned int crc32cSoftware(const unsigned char* data,
 
     // [1] Process bytes one at a time until we reach the alignment boundary
     //     (e.g. 8-byte)
-    bsls::Types::IntPtr misaligned = reinterpret_cast<bsls::Types::IntPtr>(
-                                         data) &
-                                     (sizeof(bsls::Types::IntPtr) - 1);
+    bsls::Types::UintPtr misaligned = reinterpret_cast<bsls::Types::UintPtr>(
+                                          data) &
+                                      (sizeof(bsls::Types::UintPtr) - 1);
 
     // No. of bytes from `data` to next alignment boundary
-    unsigned adj = misaligned ? sizeof(bsls::Types::IntPtr) - misaligned : 0;
+    unsigned adj = misaligned ? sizeof(bsls::Types::UintPtr) - misaligned : 0;
     if (adj > length) {
         // the specified length to process is less than the distance from
         // 'data' to the next alignment boundary.
         adj = length;
     }
 
+    // This code relies on the platform-dependent assumption that the distance
+    // from 'data' to next alignment boundary <= 7.  This is true on machines
+    // with 64-bit word sizes or smaller, but to be safe we assert this.
+    BSLS_ASSERT(adj < 8);
```

Comment:
This is probably correct, but may be the only "major" breaking change in this PR from what I see. To be clear, I'm fine with it. We'll revisit this code in the immediate near future with #688 (or similar, based on BDE's implementation).